students/JasneetChandok/Lesson04/fileexcercise.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3.6
+
+'''Lesson 4: File Excercise'''
+
+import os
+import glob
+import pathlib
+
+# Path and File Processing
+
+# Part 1
+folder_path = os.getcwd()
+list_of_file = glob.glob(os.path.join(folder_path, '*.py'))
+
+for file_name in list_of_file:
+    print(file_name)
+
+
+# Part 2
+path = pathlib.Path.cwd()
+sourcefile_path = 'C:\\Users\\jchando1\\OneDrive - T-Mobile USA\\Programming in Python\\SP_2019_210A_classroom\\students\\JasneetChandok\\Readme.txt'
+outfile_path = path / "outfile.txt"
+
+
+with open(sourcefile_path, 'rb') as infile, open(outfile_path, 'wb') as outfile:
+    outfile.write(infile.read())
+    print("File was copied to the location shared")
+
+# File reading and parsing
+distinct_langs = set()
+path2 = path / "students.txt" # directing the path to your student directory
+with open(path2) as stdfile:
+    stdfile.readline()
+    for rows in stdfile:
+        langs = rows.split(':')[1]  
+        langs = langs.replace(',', ' ')
+        langs = langs.replace('and', ' ')
+        langs = langs.split()
+        for lang in langs:
+            if lang:

```

Comment:
You are not using any logic here to filter out the nickname of the student. This will still grab the nickname if it exists. Notice that the nicknames are capitalized while the languages are not. Think about how you can ignore the nicknames in the list. 

Since you're pretty much there, I will give you credit but try to take a look at how you could resolve this issue!